api/class2.js

```diff
@@ -22,66 +25,211 @@ router.get('/customers', function(req, res) {
+  var id = req.params.id
+  const {
+    title,
+    first_name,
+    surname,
+    email
+  } = req.body
+  var sql = `update customers set 
+  title='${title}',
+  first_name='${first_name}',
+  surname='${surname}',
+  email='${email}' 
+  where id='${id}'`;
+
+
+  //if just part of it wants to be updated
+  var sql1 = `update customers set 
+  first_name='${first_name}'
+  where id='${id}'`;
+  db.run(sql1, [], (err, rows) => {
+    if (err) {
+      console.log('ERROR fetching from the database:', err);
+      return;
+    }
+    console.log('Request succeeded, new data fetched', rows);
+    res.status(200).json({
+      customers: rows
+    });
+  });
 });
 
 
+
+
 // get '/reservations'
 // TODO: add code here
 
+router.get('/reservations', function (req, res) {
+  // TODO: add code here
+  var sql = `select * from reservations`;
 
+  db.all(sql, [], (err, rows) => {
+    if (err) {
+      console.log('ERROR fetching from the database with surname:', err);
+      return;
+    }
+    console.log('Request succeeded, new data fetched', rows);
+    res.status(200).json({
+      reservations: rows
+    });
+  });
+});
 // get '/reservations/:id'
 // TODO: add code here
+router.get('/reservations/:id', function (req, res) {
+  var id = req.params.id;
+  var sql = `select * from reservations where id=${id}`;
+  console.log(sql);
+  db.all(sql, [], (err, rows) => {
+    if (err) {
+      console.log('ERROR fetching from the database:', err);
+      return;
+    }
+    console.log('Request succeeded, new data fetched', rows);
+    res.status(200).json({
+      reservations: rows
+    });
+  });
+});
+
+//delete '/reservations/:id'
+
+router.delete('/reservations/:id', function (req, res) {
+  var id = req.params.id;
+  var sql = `delete from reservations where id=${id}`;
+  db.all(sql, [], (err, rows) => {
+    if (err) {
+      console.log('ERROR deleting from the database:', err);
+      return;
+    }
+    console.log('Request succeeded, new data fetched', rows);
+    res.status(200).json({
+      reservations: rows
+    });
+  });
+})
 
 
-// delete '/reservations/:id'
-// TODO: add code here
 
 
 // get '/reservations/starting-on/:startDate'
 // TODO: add code here
-
+router.get('/reservations/starting-on/:check_in_date', function (req, res) {
+  var check_in = req.params.check_in_date;
+  var sql = `select * from reservations where check_in_date='${check_in}'`;
+  db.all(sql, [], (err, rows) => {
+    if (err) {
+      console.log('ERROR fetching from the database:', err);
+      return;
+    }
+    console.log('Request succeeded, new data fetched', rows);
+    res.status(200).json({
+      reservations: rows
+    });
+  });
+})
 
 // get '/reservations/active-on/:date'
 // TODO: add code here
 
+router.get('/reservations/active-on/:date', function (req, res) {
+  var date = req.params.date;
+  var sql = `select * from reservations INNER JOIN rooms
+   ON reservations.room_id=rooms.id where check_in_date <= '${date}' `
```

Comment:
This doesn't look right - we should be checking that date is after or the same as check in date and before check out date.